packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -592,7 +644,10 @@ class _PopupMenuRouteLayout extends SingleChildLayoutDelegate {
 
   @override
   bool shouldRelayout(_PopupMenuRouteLayout oldDelegate) {
-    return position != oldDelegate.position;
+    return position != oldDelegate.position
+        && selectedItemIndex != oldDelegate.selectedItemIndex
+        && textDirection != oldDelegate.textDirection
+        && listEquals(itemSizes, oldDelegate.itemSizes);
```

Comment:
Shouldn't this be something like this instead?:

```dart
return position != oldDelegate.position
        || selectedItemIndex != oldDelegate.selectedItemIndex
        || textDirection != oldDelegate.textDirection
        || !listEquals(itemSizes, oldDelegate.itemSizes)
```
Or am I just misunderstanding what `shouldRelayout` returns? Surely if any one of those things changes, then it should relayout?

(and if so, then there should probably be a test for that)

Reply:
Yes! Will add a test.